File: code/programmers/문자열압축.js
function solution(s) {
    var answer = s.length;
    var arr = [];
    var cnt =0;
    var check = []
    var min = s.length;
    
    for (var i = 1; i<Math.ceil(s.length/2)+1; i++) {
        arr[i] = [];
        for(var j =0; j<s.length; j+=i) {
           arr[i].push(s.substring(j, j+i))
        }
    }
    
    var last = "";
    
    for (var i =1; i< arr.length; i++) {
        last = arr[i][1]
        check = [];
        for (var j =1; j< arr[i].length; j++) {
            if ( arr[i][j-1] != arr[i][j]) {
                check.push(cnt);
                cnt =0;
            } else {
                cnt++
            }
        }

        if (cnt !=0){
            check.push(cnt)
        }
        console.log(check)
        cnt =0;
        
        for (var j =0; j<check.length; j++) {
            if (check.length > 0 && check[j] !=0){
                min -= i*check[j] - (check[j]+1).toString().length
            }
        }
        
        answer = Math.min(answer, min)
        min = s.length
    }
    console.log(arr)

    return answer;
}

File: programmers/문자열압축.cpp
#include <string>
#include <vector>

using namespace std;

int solution(string s) {
    int answer = 0;
    int count = 0;
    int max = 0;
    int check = 0;
    for(int i=1; i<=s.length; i++){
        for(int j=0; j<s.length; j+=i){
            if(s.substr(j.j+i)==s.substr(j+i,j+i+i)){
                count+=i;
            }
        }
        if (max<=count){
            max = count;
            check = i;
        }
        count = 0;
    }
    return answer;
}